data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/DefaultPluginMapper.java

```diff
@@ -0,0 +1,47 @@
+package org.opensearch.dataprepper.logstash.mapping;
+
+import com.amazon.dataprepper.model.configuration.PluginModel;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.opensearch.dataprepper.logstash.exception.LogstashMappingException;
+import org.opensearch.dataprepper.logstash.model.LogstashPlugin;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Converts Logstash plugin model to Data Prepper plugin model using mapping file
+ *
+ * @since 1.2
+ */
+public class DefaultPluginMapper implements LogstashPluginMapper {
+    private final ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+
+    @Override
+    public PluginModel mapPlugin(LogstashPlugin logstashPlugin) throws IOException {
+
+        InputStream inputStream = getClass().getClassLoader().getResourceAsStream(logstashPlugin.getPluginName() + ".mapping.yaml");
+        if (inputStream == null) {
+            throw new LogstashMappingException("file not found! " + logstashPlugin.getPluginName() + ".mapping.yaml");
+        }
+
+        LogstashMappingModel logstashMappingModel = objectMapper.readValue(inputStream, LogstashMappingModel.class);
```

Comment:
Please catch this `IOException` and wrap it in a `LogstashMappingException`.

Something like:

```
try {
    logstashMappingModel = objectMapper.readValue(inputStream, LogstashMappingModel.class);
} catch(IOException ex) {
    throw new LogstashMappingException("Unable to parse mapping file " + logstashPlugin.getPluginName() + ".mapping.yaml", ex);
}
```

You will need to add an overloaded constructor to `LogstashMappingException`.